tests/nfs_server_file_permissions_test.go

```diff
@@ -0,0 +1,302 @@
+			Expect(stderr).To(BeEmpty(), "when checking the "+
+				"stderr output of the `stat` command to "+
+				"check owner's UID",
+			)
+			Expect(stdout).To(Equal(uid), "when checking the "+
+				"stdout output of the `stat` command to "+
+				"check owner's UID",
+			)
+
+			stdout, stderr, err = Client.Exec([]string{"/bin/bash", "-c", "stat --printf=%g ${SHARED_DIRECTORY}"}, pod.Name, pod.Spec.Containers[0].Name, pod.Namespace)
+			Expect(err).To(BeNil(), "when exec-ing into the NFS "+
+				"server container in pod {%s} in namespace "+
+				"{%s}", pod.Name, pod.Namespace,
+			)
+			Expect(stderr).To(BeEmpty(), "when checking the "+
+				"stderr output of the `stat` command to "+
+				"check owner's GID",
+			)
+			Expect(stdout).To(Equal(gid), "when checking the "+
+				"stdout output of the `stat` command to "+
+				"check owner's GID",
+			)
+
+			stdout, stderr, err = Client.Exec([]string{"/bin/bash", "-c", "stat --printf=%04a ${SHARED_DIRECTORY}"}, pod.Name, pod.Spec.Containers[0].Name, pod.Namespace)
+			Expect(err).To(BeNil(), "when exec-ing into the NFS "+
+				"server container in pod {%s} in namespace "+
+				"{%s}", pod.Name, pod.Namespace,
+			)
+			Expect(stderr).To(BeEmpty(), "when checking the "+
+				"stderr output of the `stat` command to "+
+				"check the shared directory's file mode",
+			)
+			Expect(stdout).To(Equal(mode), "when checking the "+
+				"stdout output of the `stat` command to "+
+				"check the shared directory's file mode",
+			)
+		})
+	})
+
+	When("deployment with busybox image is created", func() {
+		It("should create a deployment and a running pod", func() {
+			By("building a deployment")
+			deployObj, err := deploy.NewBuilder().
+				WithName(deployName).
+				WithNamespace(applicationNamespace).
+				WithLabelsNew(labelselector).
+				WithSelectorMatchLabelsNew(labelselector).
+				WithPodTemplateSpecBuilder(
+					pts.NewBuilder().
+						WithLabelsNew(labelselector).
+						WithContainerBuildersNew(
+							container.NewBuilder().
+								WithName("busybox").
+								WithImage("busybox").
+								WithCommandNew(
+									[]string{
+										"sleep",
+										"3600",
+									},
+								).
+								WithVolumeMountsNew(
+									[]corev1.VolumeMount{
+										{
+											Name:      "demo-vol1",
+											MountPath: "/mnt/store1",
+										},
+									},
+								),
+						).
+						WithVolumeBuilders(
+							volume.NewBuilder().
+								WithName("demo-vol1").
+								WithPVCSource(pvcName),
+						),
+				).
+				Build()
+			Expect(err).ShouldNot(HaveOccurred(), "while building deployment %s/%s", applicationNamespace, deployName)
+
+			By("creating above deployment")
+			err = Client.createDeployment(deployObj)
+			Expect(err).To(BeNil(), "while creating deployment %s/%s", applicationNamespace, deployName)
+
+			By("verifying pod count as 1")
+			err = Client.waitForPods(applicationNamespace, label, corev1.PodRunning, 1)
+			Expect(err).To(BeNil(), "while verifying pod count")
+
```

Comment:
Q: Do we need to add a check verifying write accessibility of shared volume from busy-box?

Reply:
This will depend on the file set. The `stat` output verifies the mode set.
We can check write, read and execute status.... but this is checking the linux kernel behavior. This is beyond the scope of the NFS provisioner.

WDYT?

Reply:
Hmmm... Ok we can ignore